app/lib/rocketchat/methods/roomTypeToApiType.ts

```diff
@@ -4,17 +4,24 @@ enum ETypes {
 	Groups = 'groups'
 }
 
-export const types = {
+export const types: { [K in RoomTypes]: ApiTypes<K> } = {
 	c: ETypes.Channels,
 	d: ETypes.Im,
 	p: ETypes.Groups,
 	l: ETypes.Channels
 };
 
-// TODO: refactor this
-export type RoomTypes = keyof typeof types;
-type ApiTypes = typeof types[RoomTypes];
+export type RoomTypes = 'c' | 'd' | 'p' | 'l';
+type ApiTypes<T> = T extends 'c'
```

Comment:
If you're not using it anymore, you should delete it.

Reply:
But I'm using it [here](https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/3791/files/2837933b2c05f213b7c99d79c37405d02fb24889#diff-3cb1b18e31be6a3d839f926d67c736398aadba47bed898ff406f400242c17efcR7)